hr_governance/models/governance_circle.py

```diff
@@ -0,0 +1,638 @@
+            if rec.member_ids:
+                rec.assigned_user_ids = [
+                    Command.link(mem.user_id.id)
+                    for mem in rec.member_ids
+                    if mem.user_id
+                ]
+            elif rec.member_rel_ids:
+                rec.assigned_user_ids = [
+                    Command.link(mem.member_id.mapped("user_id").id)
+                    for mem in rec.member_rel_ids
+                    if mem.member_id.mapped("user_id")
+                ]
+            else:
+                rec.assigned_user_ids = False
+
+    def _compute_user_input_field(self, field_name, template_field_name):
+        for rec in self:
+            if not rec.type_id or not getattr(rec, field_name):
+                setattr(rec, f"user_input_{field_name}", "")
+                continue
+            elif self.env.context.get("skip_update_user_input"):
+                continue
+            from_template = html_to_inner_content(
+                getattr(rec.type_id, template_field_name)
+            )
+            existing = html_to_inner_content(getattr(rec, field_name))
+            cleaned_existing = existing.replace(from_template, "")
+            setattr(rec, f"user_input_{field_name}", cleaned_existing)
+
+    @api.depends("purpose")
+    def _compute_user_input_purpose(self):
+        self._compute_user_input_field(
+            field_name="purpose", template_field_name="purpose"
+        )
+
+    @api.depends("expectation")
+    def _compute_user_input_expectation(self):
+        self._compute_user_input_field(
+            field_name="expectation", template_field_name="expectation"
+        )
+
+    @api.depends("authority")
+    def _compute_user_input_authority(self):
+        self._compute_user_input_field(
+            field_name="authority", template_field_name="authority"
+        )
+
+    @api.depends("parent_id", "parent_id.suitable_type_ids")
+    def _compute_suitable_type_ids(self):
+        for rec in self:
+            if rec.parent_id:
+                rec.suitable_type_ids = self.env["governance.role.type"].search(
+                    [
+                        ("source_circle_id", "=", rec.parent_id.id),
+                    ]
+                )
+                rec.suitable_type_ids |= rec.parent_id.suitable_type_ids
+            else:
+                rec.suitable_type_ids = [Command.clear()]
+
+    def _compute_member_ids(self):
+        for rec in self:
+            rec.member_ids = rec.child_ids.member_rel_ids.mapped("member_id")
+
+            # include member of steering roles of its subcircles
+            # only the next level, not the whole hiearchy
+            steering_roles = rec.child_ids.child_ids.filtered(
+                lambda x: not x.is_circle and x.is_steering_role
+            )
+            rec.member_ids |= steering_roles.member_rel_ids.mapped("member_id")
+
+    @api.model
+    def get_greyscale_mode_param(self):
+        is_greyscale_mode_on = (
```

Comment:
```suggestion
        is_greyscale_mode_on = str2bool(
```

import from `odoo.tools.misc`